### PR TITLE
fix: Fix support for @auth on fields with no @model

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

Fix "missing datasource" issue reported in #2133, due to regression in support for `@auth` directives on root types.

Do not merge this until we resolve the correct branch -- this might involve reverting `main` to the post-release state so we can apply the hotfix, unless we want to craft a new release automation to apply to a different branch.

#### Issue #, if available

#2133 

#### Description of how you validated changes

- Added new tests to cover V2 transformer support for this case
- [Rerunning existing E2E](https://us-east-1.console.aws.amazon.com/codesuite/codebuild/594813022831/projects/amplify-category-api-e2e-workflow/batch/amplify-category-api-e2e-workflow:b1adb08c-716b-46b0-a166-52665cb1892d?region=us-east-1) tests to ensure no breakage

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-auth-transformer/src/__tests__/NonModelAuthTransformer.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { GraphQLTransform } from 'graphql-transformer-core';
```
Comment (palpatim) [inline packages/graphql-auth-transformer/src/__tests__/NonModelAuthTransformer.test.ts:1]: These are just some linting cleanups of the V1 transformer test that I applied while I was porting the tests to the V2 transformer

File: packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
```
Comment (palpatim) [inline packages/amplify-graphql-auth-transformer/src/__tests__/non-model-auth.test.ts:1]: This ports the V1 transformer tests that covered `@auth` directives on root types. It should cover the same use cases as the V1 test, although I slightly reworded the `it` clauses to more clearly (according to me) reflect what's being asserted.

**Review by phani-srikar — APPROVED** [review]

**Review by AaronZyLee — APPROVED** [review]
LGTM. If the hot fix is to be merged to `release` branch and released there, I suggest a backup for the `main` branch before you make force push on `main`. After the hot fix is merged back to main, the previous not released commits can be cherry-picked from the backup branch.

**Comment by palpatim** [comment]
> LGTM. If the hot fix is to be merged to `release` branch and released there, I suggest a backup for the `main` branch before you make force push on `main`. After the hot fix is merged back to main, the previous not released commits can be cherry-picked from the backup branch.

No force-push necessary: I experimented with what I believe would be the commit flows, and everything merges cleanly.

That said, I'll definitely be making a backup of main, just to be on the safe side!

**Review by dpilch — APPROVED** [review]

**Review by iartemiev — APPROVED** [review]

**Pull request merged**